lib/Optimizer/Transforms/GenKernelExecution.cpp

```diff
@@ -268,53 +351,78 @@ class GenerateKernelExecution
     Value zero = builder.create<arith::ConstantIntOp>(loc, 0, 64);
     Value extraBytes = zero;
 
-    // Loop over the struct elements
+    // Process all the arguments for the original call by looping over the
+    // kernel's arguments.
     bool hasTrailingData = false;
-    for (auto structElementTypeIter : llvm::enumerate(structTy.getMembers())) {
-      std::int64_t idx = structElementTypeIter.index();
+    for (auto kaIter : llvm::enumerate(kernelArgTypes)) {
+      std::int64_t idx = kaIter.index();
```

Comment:
Maybe this is related to the other question I asked, but why not `std::size_t` here?

Reply:
I should probably make them all int32_t. Let me check again.